Firestore/Source/Local/FSTPersistence.h

```diff
@@ -113,26 +114,47 @@ struct FSTTransactionRunner;
 @end
 
 struct FSTTransactionRunner {
-
 // Intentionally disable nullability checking for this function. We cannot properly annotate
 // the function because this function can handle both pointer and non-pointer types. It is an error
 // to annotate non-pointer types with a nullability annotation.
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wnullability-completeness"
   template <typename F>
-  auto operator() (F block) const -> decltype(block()) {
+  auto operator()(F block) ->
+      typename std::enable_if<std::is_void<decltype(block())>::value, void>::type {
+    __strong id<FSTTransactional> strongDb = _db;
+    if (!strongDb && _expect_db) {
+      FSTCFail(@"Transaction runner accessed without underlying db when it expected one");
+    }
+    if (strongDb) {
```

Comment:
It doesn't seem like we should do anything if the _db is gone.

```objc
if (!strongDb) {
  return;
}
```

or, in the typed case:

```objc
if (!strongDb) {
  return ReturnT();
}
```

Reply:
`_db` is never present for memory-backed persistence.